freva-rest/src/freva_rest/stac_api/core.py

```diff
@@ -0,0 +1,750 @@
+import json
+import sys
+from textwrap import dedent
+from typing import Any, AsyncGenerator, Dict, List, Literal, Optional, cast
+from urllib.parse import urlencode
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from freva_rest.config import ServerConfig
+from freva_rest.databrowser_api import STAC, Asset, Item, Link, Solr
+from freva_rest.logger import logger
+
+from .schema import (
+    CONFORMANCE_URLS,
+    STAC_VERSION,
+    STACCollection,
+    STACItemCollection,
+    STACLinks,
+)
+
+
+class STACAPI:
+    """STAC API implementation for the Freva with Solr Backend."""
+
+    def __init__(
+        self,
+        config: ServerConfig,
+        *,
+        limit: int = 12,
+        token: Optional[str] = None,
+        datetime: Optional[str] = None,
+        bbox: Optional[str] = None,
+        uniuq_key: Literal["file", "uri"] = "file",
+        **query: list[str],
+    ) -> None:
+        self.config = config
+        self.uniq_key = uniuq_key
+        self.solr_object = Solr(config)
+        self.stac_object = STAC(config)
+        self.solr_query = self.solr_object.query
```

Comment:
Could you define search method in the solr core object that you could call here, instead of mingling with the query dict?

If we really want to use different search back ends you then this class shouldn't know about the search backend it is using.